packages/ui/src/components/shadcn/ui/input.tsx

```diff
@@ -18,12 +18,32 @@ const InputVariants = cva('aria-[]', {
   },
 })
 
-const Input = React.forwardRef<HTMLInputElement, InputProps>(
+const Input = forwardRef<HTMLInputElement, InputProps>(
   ({ className, type, size = 'small', ...props }, ref) => {
+    const inputRef = useRef<HTMLInputElement>(null)
+
+    useImperativeHandle(ref, () => inputRef.current!)
```

Comment:
🤯 so much better than the shenanigans I do to merge refs, will be copying this trick 🧠